### PR TITLE
Fix vect parsing with newline before comma

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/JuliaLang/JuliaSyntax.jl/issues/59.

### CODE REVIEW

**Comment by oscardssmith** [comment]
Looks good. Merge?

**Comment by pfitzseb** [comment]
I'll merge this tomorrow morning, unless Chis chimes in and says this is the wrong fix :)

The test is currently a bit awkward, because SyntaxNode/Expr conversion can't handle unexpected `,`s. Would like to fix that.

File: test/parser.jl [hunk 1]
```diff
@@ -779,6 +782,10 @@ broken_tests = [
     ]
 ]
 
+@testset "Invalid syntax" begin
+    @test !isempty(JuliaSyntax.parse(JuliaSyntax.GreenNode, """[ [],,[] ]""")[2])
```
Comment (c42f) [inline test/parser.jl:786]: Let's make a more precise test case out of this and add it to the vector of `tests`.

The parser should emit some `error` node and we can test that.

File: src/parser.jl [hunk 2]
```diff
@@ -2798,7 +2798,7 @@ function parse_cat(ps::ParseState, closer, end_is_symbol)
     if k == K"," || (is_closing_token(ps, k) && k != K";")
         if k == K","
             # [x,]  ==>  (vect x)
-            bump(ps, TRIVIA_FLAG)
+            bump(ps, TRIVIA_FLAG; skip_newlines = true)
```
Comment (c42f) [inline src/parser.jl:2801]: Cool looks good. I wonder how many more cases like this there are elsewhere in the parser ... I've found a few before ...

**Review by c42f — COMMENTED** [review]
Looks good thanks.  Let's add such error cases to the tests as s-expressions along with the happy cases.

**Comment by c42f** [comment]
> The test is currently a bit awkward, because SyntaxNode/Expr conversion can't handle unexpected

Ah yeah I didn't see this comment before I wrote the review.  Precisely :)

**Comment by pfitzseb** [comment]
Tests look a lot nicer now. We're now just emitting an `ErrorVal` when encountering unknown node kinds, which may be a bad idea, but at least allows us to recover from the corrupted tree we produce further down :P

**Comment by pfitzseb** [comment]
This good to go, or do you think emitting `ErrorVal`s instead of actually erroring is going to be an issue further down the line?

**Review by c42f — APPROVED** [review]
The FIXME with `ErrorVal` makes me a bit uncomfortable (does seem wrong). But this is still an improvement so I think we can merge and fix that later.

**Pull request merged**